src/main/java/frc/robot/Orchestrator.java

```diff
@@ -0,0 +1,128 @@
+      RobotState robotState) {
+    this.drive = drive;
+    this.intake = intake;
+    this.indexer = indexer;
+    this.shooter = shooter;
+    this.pixy2 = pixy2;
+    this.arm = arm;
+    this.robotState = robotState;
+  }
+
+  public Command turnToSpeaker() {
+    Supplier<Pose2d> targetPose =
+        () ->
+            new Pose2d(
+                drive.getPose().getTranslation(),
+                speaker.minus(drive.getPose().getTranslation()).getAngle());
+    Logger.recordOutput("Autos/Speaker", speaker);
+    return Commands.defer(() -> new StraightDriveToPose(targetPose.get(), drive), Set.of(drive));
+  }
+
+  public Command driveToNote(Translation2d targetTranslation) {
+    Supplier<Rotation2d> targetRotation =
+        () -> targetTranslation.minus(drive.getPose().getTranslation()).getAngle();
+    return Commands.defer(
+        () -> new StraightDriveToPose(new Pose2d(targetTranslation, targetRotation.get()), drive),
+        Set.of(drive));
+  }
+
+  public Command alignArm() {
+    return Commands.defer(
+        () ->
+            arm.toSetpoint(
+                new Rotation2d(
+                    shooter.calculateShootingAngle(
+                            drive.getPose().getTranslation().getDistance(speaker))
+                        - ArmConstants.HORIZONTAL_OFFSET.getRadians())),
+        Set.of(arm));
+  }
+
+  public Command shootBasic() {
+    return Commands.sequence(
+            Commands.parallel(
+                shooter.shoot(9999)
+                // Turn towards speaker.
+                ),
+            Commands.waitUntil(shooter::ShooterSpeedIsReady).withTimeout(2),
+            indexer.setVolts(1),
+            Commands.waitUntil(() -> !indexer.getLimitSwitchPressed()).withTimeout(2),
+            Commands.parallel(indexer.setVolts(0), shooter.manualShoot(0)))
+        .onlyIf(indexer::getLimitSwitchPressed);
+  }
+
+  public Command intakeBasic() {
+    return Commands.sequence(
+        indexer.setVolts(4),
+        arm.toSetpoint(ArmConstants.HORIZONTAL_OFFSET), // TODO: Make setPoint for pickup position.
+        Commands.waitUntil(arm::atSetpoint).withTimeout(2),
+            intake.intake().until(() -> robotState.hasNote));
+  }
+
+  public Command driveWhileIntaking() {
+    return Commands.parallel(intake.intake().until(() -> robotState.hasNote), Commands.run(()->drive.runVelocity(new ChassisSpeeds(Units.inchesToMeters(10), 0.0,0.0)),drive).withTimeout(2));
+  }
+
+  // Intakes after seeing note with Pixy2.
+  public Command visionIntake() {
+    return Commands.sequence(
+        indexer.setVolts(4),
+        arm.toSetpoint(ArmConstants.HORIZONTAL_OFFSET), // TODO: Make setpoint for pickup position.
+        Commands.waitUntil(() -> arm.atSetpoint() && pixy2.seesNote()).withTimeout(2),
+        intake.intake().until(() -> robotState.hasNote));
+
+  }
+
+  public Command expelFullRobot() {
+    return Commands.parallel(
+        shooter.manualShoot(-5), indexer.setVolts(-5.0), intake.release());
```

Comment:
do expelShindex and expelIntake in parallel rather than this

Reply:
Fixed